verifiable/onchain/parser.go

```diff
@@ -0,0 +1,473 @@
+package onchain
+
+import (
+	"encoding/hex"
+	"encoding/json"
+	"fmt"
+	"math/big"
+	"strconv"
+	"strings"
+	"time"
+
+	"github.com/google/uuid"
+	core "github.com/iden3/go-iden3-core/v2"
+	"github.com/iden3/go-iden3-core/v2/w3c"
+	"github.com/iden3/go-merkletree-sql/v2"
+	"github.com/iden3/go-schema-processor/v2/merklize"
+	"github.com/iden3/go-schema-processor/v2/verifiable"
+	"github.com/piprate/json-gold/ld"
+)
+
+type Parser struct {
```

Comment:
I would avoid term Parser due to the collision with JSON / JSONLD parsers. better use DataConvertor. This convertor serves only single purpose.